### DISCUSSION TITLE
ERROR with BT-765-Lot and BT-766-Lot

### DISCUSSION THREAD

**Discussion by user6330** [OP-TED/eForms-SDK#580] [Q&A] [answered]
I have **BT-765-Lot** and **BT-766-Lot** both mandatory for noticeype.

During the creation of XML 

![image](https://github.com/OP-TED/eForms-SDK/assets/127392893/c58747ac-3168-488c-84a5-b545ee29928b)

the XSD **UBL-ContractNotice-2.3.xsd** gave me validation xml structure error because ContractingSystemTypeCode has maxOccurs="1"

![image](https://github.com/OP-TED/eForms-SDK/assets/127392893/3239a124-12a8-4aa1-b9b5-8e89b6243e5e)

Can You solve this?

**Comment by YvesJo** [comment]
Hi,
There is indeed an issue with the XML generation, **framework agreement** and **dps usage** should occur in different parents (i.e. different occurrences of a same element) as illustrated below (cf. notice example can_24_FRA_comments.xml):
```
		<cac:TenderingProcess>
			<!-- removed mark-up -->
			<cac:ContractingSystem>
				<!--BT-765 - Framework Agreement-->
				<cbc:ContractingSystemTypeCode listName="framework-agreement">none</cbc:ContractingSystemTypeCode>
			</cac:ContractingSystem>
			<cac:ContractingSystem>
				<!--BT-766 - Dynamic Purchasing System-->
				<cbc:ContractingSystemTypeCode listName="dps-usage">none</cbc:ContractingSystemTypeCode>
			</cac:ContractingSystem>
		</cac:TenderingProcess>
```
Solving this requires an update of the XML generator you are using

KR

**Reply by user6330** [reply, marked as answer]
Hi thanks for your reply.

I'm using the notice-editor, so it's wrong their implementation

**Reply by YvesJo** [reply]
The notice editor demo is not fully operational; it is a side development to illustrate what is feasible.  I can't tell if some evolution is foreseen in the short term.

**Reply by user6330** [reply]
How can I understand that **framework agreement** and **dps usage** should occur in different parents?

**Reply by YvesJo** [reply]
there are multiple sources available to lead to that conclusion:
- the online documentation with the markup provided excerpts,
- the notice samples provided with the SDK,
- the constraints imposed by the XSD (cbc:ContractingSystemTypeCode is not repeatable but its parent is)
To further help with that, in a next SDK release, additional nodes have been defined and NTDs updated accordingly.

**Reply by user6330** [reply]
So there is nothing written that precisely defines the logic but inevitably it is the only solution.

Thank you
